axlearn/common/attention.py

```diff
@@ -1956,9 +1956,10 @@ def _forward_for_mode(
                     key_positions = query_positions
                     kv_state = KVState(k_proj, v_proj, key_positions)
             else:
-                # KV sharing branch.
-                k_proj, v_proj, key_positions = kv_state
-                kv_state = KVState(*kv_state)
+                # KV sharing branch. Pack the `k_proj` and `v_proj` (possibly
+                # updated by i_proj), and the same `key_positions`.
+                key_positions = kv_state.key_positions
```

Comment:
```suggestion
                # KV sharing branch.
                # 
                # When we call `i_proj` above, we already pass `kv_state` to it, so that i_proj will use shared KV and update it.
                # 
                # Here we pack the `k_proj` and `v_proj` (possibly
                # updated by i_proj), and the same `key_positions`, into `kv_state`.
                key_positions = kv_state.key_positions
```